src/commands.ts

```diff
@@ -197,10 +198,56 @@ export class Commands<C extends Context> {
     }
 
     /**
-     * Alias for {@link toArgs}
+     * Serialize all register commands into an array of {@link CommandElementals},
+     * each registered command will be summarized into it's name, prefix and language
+     *
+     * @param filterLanguage If given and valid, filters out all command localizations
+     * that not match the filterLanguage,
+     * if a command does not have the given language, it would fallback to "default"
+     *
+     * Note: mainly used to serialize for {@link FuzzyMatch}
      */
-    public toJSON() {
-        return this.toArgs();
+
+    public toElementals(filterLanguage?: string): CommandElementals[] {
```

Comment:
Consider:
- Simplify into a single implementation instead of relying into the current branching approach. Return values must remain the same, ensured by the current test's